### PR TITLE
feat(*): modules export entities and fields

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
  Modules will export their Entities and Fields available using GraphQL Schema Definition Language.
  
Why:
  When multiple modules are loaded, their relationships and their available fields have to be accessible by MedusaApp.

How:
  The public method `__joinerConfig` of the module has to export a property `schema` containing the string definition of the schema.
  
  ```typescript
  export const joinerConfig: ModuleJoinerConfig = {
    serviceName: Modules.PRODUCT,
    linkableKeys: {
      product_id: "Product",
      product_handle: "Product",
      variant_id: "ProductVariant",
      variant_sku: "ProductVariant",
    },
    schema: `
      scalar JSON
      type Product {
        id: ID!
        title: String!
        handle: String
        description: String
        variants: [ProductVariant]
        metadata: JSON
      }
      
      type ProductVariant {
        id: ID!
        title: String!
        sku: String
        variantRank: Float
        productId: String!
        product: Product!
      }
    `,
   }
  ```

### CODE REVIEW

**Comment by olivermrbl** [comment]
@carlos-r-l-rodrigues – can I get you to add a PR description of what's included? For the sake of making it easier for Shahed to document in the near future? Doesn't have to be fully fleshed out; high-level bullets covering what, why, and how should suffice.

File: .changeset/plenty-bugs-push.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
```
Comment (olivermrbl) [inline .changeset/plenty-bugs-push.md:1]: question: nitpicking, but should this be a minor? Introduces a couple of minor breaking changes I believe

File: packages/modules-sdk/src/utils/clean-graphql-schema.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      typeNames.add(def.name.value)
+    } else if (def.kind === Kind.OBJECT_TYPE_EXTENSION) {
+      extendedTypes.add(def.name.value)
+    }
+  })
+
+  const nonExistingMap: Record<string, Record<string, string>> = {}
+  const parentStack: string[] = []
+
+  const cleanedAst = visit(ast, {
```
Comment (olivermrbl) [inline packages/modules-sdk/src/utils/clean-graphql-schema.ts:45]: comment: looks very clean; could you add a couple of comments in code here, describing the work being done? Think that would be helpful for the future

**Review by olivermrbl — APPROVED** [review]
Looks really great! Added a couple of comments. 

We could add some tests of the Graphql tooling

File: .changeset/plenty-bugs-push.md (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply .changeset/plenty-bugs-push.md:1]: true, the format of the linkableKeys has changed.

Comment (adrien2p) [reply .changeset/plenty-bugs-push.md:1]: I don't think it is an issue, I doubt any body has created links yet and we have migrated all of them. but go for a minor

Comment (carlos-r-l-rodrigues) [reply .changeset/plenty-bugs-push.md:1]: is the changeset filename a premonition? 😆

Comment (olivermrbl) [reply .changeset/plenty-bugs-push.md:1]: Hahaha, that's an unfortunate name 😅

**Comment by carlos-r-l-rodrigues** [comment]
> @carlos-r-l-rodrigues – can I get you to add a PR description of what's included? For the sake of making it easier for Shahed to document in the near future? Doesn't have to be fully fleshed out; high-level bullets covering what, why, and how should suffice.

@olivermrbl @shahednasser let me know if that is enough or we need it more detailed.

**Comment by shahednasser** [comment]
@carlos-r-l-rodrigues looks good, thanks!

**Comment by olivermrbl** [comment]
> let me know if that is enough or we need it more detailed.

Super nice, thanks!

**Comment by olivermrbl** [comment]
@carlos-r-l-rodrigues – should we merge? 🚀

**Comment by carlos-r-l-rodrigues** [comment]
> @carlos-r-l-rodrigues – should we merge? 🚀

Sure. We can merge it.

**Pull request merged**